src/services/utilities.ts

```diff
@@ -1086,6 +1086,102 @@ namespace ts {
             return !seen[id] && (seen[id] = true);
         };
     }
+
+
+    const enum PromiseState { Unresolved, Success, Failure }
+    /**
+     * Cheap PromiseLike implementation.
+     */
+    export class PromiseImpl<T> implements PromiseLike<T> {

```

Comment:
Why do we need a custom Promise implementation?

Reply:
node v4.0 seems like a reasonable minimum supported platform for the services..

Reply:
@anstarovoyt does your engine have Promise support?

Reply:
we don't use it

Reply:
I am sorry, do not use what?

I am asking about your runtime, is it node v4 or compatible?

Reply:
yes, we expect that our users use node  >= v4 for running the service